src/asm/lexer.c

```diff
@@ -2288,23 +2292,19 @@ int yylex(void)
 	int token = lexerModeFuncs[lexerState->mode]();
 
 	if (token == T_EOF) {
-		if (lexerState->lastToken != T_NEWLINE) {
-			dbgPrint("Forcing EOL at EOF\n");
-			token = T_NEWLINE;
-		} else {
-			/* Try to switch to new buffer; if it succeeds, scan again */
-			dbgPrint("Reached EOF!\n");
-			/* Captures end at their buffer's boundary no matter what */
-			if (!lexerState->capturing) {
-				if (!yywrap())
-					goto restart;
+		/* Try to switch to new buffer; if it succeeds, scan again */
+		dbgPrint("Reached EOB!\n");
+		/* Captures end at their buffer's boundary no matter what */
+		if (!lexerState->capturing) {
+			if (yywrap()) {
```

Comment:
I think that this check could instead be moved to the beginning of `yylex`, with the condition changed to `lexerState->lastToken == T_EOB`, thereby removing the need for `lexerState->isAtEOF`. What do you think?

Reply:
I think that works. Will see if it passes all tests.